.github/workflows/main.yml

```diff
@@ -111,7 +111,7 @@ jobs:
         gpg-passphrase: MAVEN_GPG_PASSPHRASE
 
     - name: Build the nar and publish
-      run: mvn --batch-mode nar:nar-prepare-package nar:nar-package jar:jar deploy:deploy
+      run: mvn --batch-mode nar:nar-prepare-package nar:nar-package jar:jar gpg:sign deploy:deploy
```

Comment:
You probably also need to enable the "sonatype-deploy" profile: `-Psonatype-deploy`

Reply:
It seems this wasn't needed (which mean the profile can be removed altogether).

Reply:
(Only note that the version of gpg-maven-plugin that will be used for gpg:sign will be undetermined.)